cmd/restic/cmd_check.go

```diff
@@ -316,16 +336,18 @@ func runCheck(ctx context.Context, opts CheckOptions, gopts GlobalOptions, args
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		bar := newTerminalProgressMax(!gopts.Quiet, 0, "snapshots", term)
+		bar := newTerminalProgressMax(!gopts.Quiet && !gopts.JSON, 0, "snapshots", term)
 		defer bar.Done()
 		chkr.Structure(ctx, bar, errChan)
 	}()
 
+	result.NumTreeErrors = 0
 	for err := range errChan {
 		errorsFound = true
 		if e, ok := err.(*checker.TreeError); ok {
 			printer.E("error for tree %v:\n", e.ID.Str())
 			for _, treeErr := range e.Errors {
+				result.NumTreeErrors++
 				printer.E("  %v\n", treeErr)
 			}
 		} else {
```

Comment:
The `printer.E` below is not accounted for.

Reply:
I now also increment the unified errors count in the `else` clause.

Would this be the "snapshot error" type as mentioned in https://github.com/restic/restic/pull/5194#discussion_r1938346379?

Reply:
Those error include the "snapshot error" type, but probably also some other errors.